custom/inddex/food.py

```diff
@@ -188,67 +228,81 @@ def age_range(self):
             return "50-64 years"
         return "65+ years"
 
-    def as_list(self):
-
-        def _format(val):
-            if isinstance(val, datetime):
-                return val.strftime('%Y-%m-%d %H:%M:%S')
-            if isinstance(val, bool):
-                return "yes" if val else "no"
-            if isinstance(val, int):
-                return str(val)
-            if isinstance(val, float):
-                return str(int(val)) if val.is_integer() else str(val)
-            if val is None:
-                return MISSING
-            return val
-
-        return [_format(getattr(self, column.slug)) for column in INDICATORS]
-
-
-class FoodCaseRow(FoodRow):
-    """A food item directly corresponding to a case in the UCR"""
-
-    def __init__(self, ucr_row, fixtures):
-        super().__init__(ucr_row['food_code'], ucr_row, fixtures)
-        self.caseid = ucr_row['doc_id']
+    @property
+    def recipe_id(self):
+        if self.is_recipe:
+            return self.caseid
+        return self.recipe_case_id or self.already_reported_recipe_case_id or 'NO_RECIPE'
 
     def __getattr__(self, name):
-        # If it's an indicator in the UCR that hasn't been explicitly set, return that val
         if name in _INDICATORS_BY_SLUG:
-            indicator = _INDICATORS_BY_SLUG[name]
-            return self.ucr_row[indicator.slug] if indicator.in_ucr else None
-        return super().__getattr__(name)
-
-
-class RecipeIngredientRow(FoodRow):
-    """A food item inferred from a recipe"""
-    is_ingredient = "yes"
-
-    def __init__(self, ucr_row, fixtures, ingredient):
-        # ucr_row is data from the parent food case
-        # ingredient is static info for this ingredient from the recipes fixture
-        super().__init__(ingredient.ingr_code, ucr_row, fixtures)
+            if self._is_std_recipe_ingredient:
+                # If it's an indicator that hasn't been explicitly set, check if it can
+                # be pulled from the food fixture or from the parent food case's UCR
+                indicator = _INDICATORS_BY_SLUG[name]
+                if indicator.in_food_fixture:
+                    return getattr(self.fixtures.foods[self.food_code], indicator.slug)
+                if indicator.is_recall_meta:
```

Comment:
I forgot to ask, what is "recall meta"?

Reply:
A poor choice of variable name.

When a user asks a beneficiary about their eating habits, that's a food recall session, resulting in a "foodrecall" case.  Each row in this report has a parent food recall.  My intent with the name was to say something like, "this is metadata about the parent case".  Okay, so what?  That means it applies to all rows resulting from the recall, whether food cases or ingredient rows pulled from the fixture.  Dunno, still can't think of a more obvious term, any ideas?

Reply:
Thanks. Now I understand. I think as soon as someone knows that "recall" is the parent case, the name makes sense. I guess you could rename it to "is_parent_case_meta", but that's longer, so I think "is_recall_meta" is fine. ... Or you could go for "is_meta_about_memories_of_food". :stuck_out_tongue_closed_eyes: